runtime/vm/createramclass.cpp

```diff
@@ -2403,9 +2402,15 @@ internalCreateRAMClassFromROMClassImpl(J9VMThread *vmThread, J9ClassLoader *clas
 
 			/* flattened classes cache */
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			UDATA flattenedClassCacheAllocSize;
+			if (J9_ARE_ALL_BITS_SET(romClass->modifiers, J9AccValueType) || (flattenedClassCache->numberOfEntries > 0)) {
+				flattenedClassCacheAllocSize = SIZEOF_FLATTENED_CLASS_CACHE_HEADER + sizeof(J9FlattenedClassCacheEntry) * (flattenedClassCache->numberOfEntries);
+			} else {
```

Comment:
the else case is not needed. If the class doesnt contain flattened fields or it is not a valuetype no memory should be allocated for the flattened class cache